### DISCUSSION TITLE
how to hint only link in a specific area of the screen?

### DISCUSSION THREAD

**Discussion by almostnobody** [tridactyl/tridactyl#3987] [Q&A] [answered]
I want to be able to create hints for links near mouse cursor/in a specific area of the screen/any page, Is it possible without modifying tridactyl code?
If not, in which part of tridactyl code structure should i implement it?

**Comment by mozbugbox** [comment, marked as answer]
You can look into the `hint()` function at `excmds.ts` to see how hint works. and create your own excmd.

https://github.com/tridactyl/tridactyl/blob/b3c059fff674cc2c524ae0689de6890b1596ccff/src/excmds.ts#L4680-L4683

In practice, the `hint()` calls the `hintPage()` to do its job. You can provide a `hintables`, which contains only links near the mouse pointer, to the `hintPage()` action.

The `hintables` are implemented in `lib/hint_util.ts`.

**Comment by almostnobody** [comment]
Thanks, seems reasonably simple
